ui/StatusQ/src/leftjoinmodel.cpp

```diff
@@ -23,7 +23,35 @@ void LeftJoinModel::initialize(bool reset)
     auto rightRoleNames = m_rightModel->roleNames();
 
     auto leftNames = leftRoleNames.values();
-    auto rightNames = rightRoleNames.values();
+    QList<QByteArray> rightNames;
+
+    if (m_rolesToJoin.empty()) {
+        rightNames = rightRoleNames.values();
+    } else {
+        QHash<int, QByteArray> tmpRightRoleNames;
+
+        auto rolesToJoin = m_rolesToJoin;
+
+        if (rolesToJoin.indexOf(m_joinRole) == -1)
+            rolesToJoin << m_joinRole;
+
+        for (auto& roleName : qAsConst(rolesToJoin)) {
+            auto name = roleName.toUtf8();
+            auto roles = rightRoleNames.keys(name);
+
+            if (roles.empty()) {
+                qWarning().noquote()
+                        << QString("Role to join %1 not found in the right model!")
+                           .arg(roleName);
+                return;
```

Comment:
Shouldn't it just `continue` here with the next role?

Reply:
Yeah, I think that both variants are ok. This one is more strict and kind of fail-fast approach. If you declared roles it's presumed that you really need them, otherwise proxy won't be initialized.

Reply:
I'd still like to see the `continue` here, just to be more resilitent to backend model changes (e.g. when an unused role gets dropped)

Reply:
I don't perceive it as resilience - rather easy way to overlook subtle bugs. If model is not initialized it should be easily detected by e.g. high-level squish tests bc it won't be possible to set channels and complete permission flow using channels. If the role is silently skipped it may easily introduce long-lasting small bug in the ui like e.g. incorrect names rendering.

I can change it if you insist but please consider all pros/cons here :)

Reply:
Ok, fair enough, I think it's good to stay on the conservative side 😊